pageserver/src/tenant/timeline.rs

```diff
@@ -3130,37 +3263,39 @@ impl Timeline {
             }
         }
 
+        let child_ctx = ctx.register_another(RequestContext::new(
+            TaskKind::DownloadAllRemoteLayers,
+            DownloadBehavior::Download,
+        ));
+        let task_id = child_ctx.context_id();
```

Comment:
I changed this to use 'context_id' instead of 'task_id', because I removed the concept of a 'task_i'. It's not entirely clear to me what the purpose of the task_id in `spawn_download_all_remote_layers` is however. Can you take a look and clarify that please, @problame?

Probably should rename 'task_id' at the very least here..